pghoard/rohmu/object_storage/sftp.py

```diff
@@ -0,0 +1,209 @@
+
+    def get_contents_to_fileobj(self, key, fileobj_to_store_to, *, progress_callback=None):
+        self._get_contents_to_fileobj(key, fileobj_to_store_to, progress_callback)
+        return self.get_metadata_for_key(key)
+
+    def _get_contents_to_fileobj(self, key, fileobj_to_store_to, progress_callback=None):
+        target_path = self.format_key_for_backend(key.strip("/"))
+        self.log.debug("Get file content: %r", target_path)
+
+        try:
+            return self.client.getfo(target_path, fileobj_to_store_to, progress_callback)
+        except FileNotFoundError as ex:
+            raise FileNotFoundFromStorageError(key) from ex
+
+    def get_contents_to_string(self, key):
+        bio = BytesIO()
+        metadata = self.get_contents_to_fileobj(key, bio)
+        return bio.getvalue(), metadata
+
+    def get_file_size(self, key):
+        target_path = self.format_key_for_backend(key.strip("/"))
+        try:
+            return self.client.stat(target_path).st_size
+        except FileNotFoundError as ex:
+            raise FileNotFoundFromStorageError(key) from ex
+
+    def get_metadata_for_key(self, key):
+        bio = BytesIO()
+        self._get_contents_to_fileobj(key + ".metadata", bio)
+        return json.loads(bio.getvalue().decode())
+
+    @staticmethod
+    def _skip_file_name(file_name):
+        return file_name.startswith(".") or file_name.endswith(".metadata") or ".metadata_tmp" in file_name
+
+    def iter_key(self, key, *, with_metadata=True, deep=False, include_key=False):
+        target_path = self.format_key_for_backend(key.strip("/"))
+        self.log.debug("Listing path: %r", target_path)
+
+        try:
+            attrs = self.client.listdir_attr(target_path)
+        except FileNotFoundError:  # if not a directory will throw exception
+            if include_key:
+                file_name = os.path.basename(target_path)
+                if self._skip_file_name(file_name):
+                    return
+
+                try:
+                    attr = self.client.stat(target_path)
+
+                    if with_metadata:
+                        metadata = self.get_metadata_for_key(key)
+                    else:
+                        metadata = None
+
+                    last_modified = datetime.datetime.fromtimestamp(attr.st_mtime, tz=datetime.timezone.utc)
+                    yield IterKeyItem(
+                        type=KEY_TYPE_OBJECT,
+                        value={
+                            "name": key,
+                            "size": attr.st_size,
+                            "last_modified": last_modified,
+                            "metadata": metadata,
+                        }
+                    )
+                    return
+                except FileNotFoundError:
+                    return
+
+            else:
+                return
+
+        files = []
+        for attr in attrs:
+            files.append(attr.filename)
```

Comment:
Could use comprehension and since this is used up for doing `in` checks using `set` would work much better. `files = set(attr.filename for attr in attrs)`